tests/test_examples/test_runs_via_configs/test_via_configs.py

```diff
@@ -59,3 +63,99 @@ def test_train_with_categories(accelerator: str, devices: int) -> None:
+    command = (
+        f"inference_images_mock.py ~dataframe_name dataset_root={MOCK_DATASET_PATH} " f"features_file={features_path}"
+    )
+    cfg_path = SCRIPTS_PATH / "configs" / "inference_images_mock.yaml"
+    yield command, features_path, cfg_path
+    features_path.unlink()
+
+
+@pytest.fixture()
+def features_from_df_command_and_paths(
+    accelerator: str, devices: int, cleanup: bool = True
+) -> Iterator[Tuple[str, Path, Path]]:
+    cfg_path = SCRIPTS_PATH / "configs" / "inference_images_mock.yaml"
+    features_path = MOCK_DATASET_PATH / "features2.json"
+
+    command_df_part = "inference_images_mock.py dataframe_name={df_name} "
+    command_features_path_part = f"dataset_root={MOCK_DATASET_PATH} ~images_folder features_file={str(features_path)}"
+    command = command_df_part + command_features_path_part
+    yield command, features_path, cfg_path
+    features_path.unlink()
+
+
+@pytest.mark.parametrize("accelerator, devices", accelerator_devices_pairs())
+def test_inference_images_list(
+    accelerator: str, devices: int, features_from_path_command_and_paths: Tuple[str, Path, Path]
+) -> None:
+    command, features_path, cfg_path = features_from_path_command_and_paths
+    run(command, accelerator, devices, cfg_path=cfg_path)
+
+
+@pytest.mark.parametrize("accelerator, devices", accelerator_devices_pairs())
+def test_inference_dataframe_w_boxes(
+    accelerator: str, devices: int, features_from_df_command_and_paths: Tuple[str, Path, Path]
+) -> None:
+    command, features_path, cfg_path = features_from_df_command_and_paths
+    command = command.format(df_name=MOCK_DATASET_PATH / "df_with_bboxes.csv")
+    run(
+        command,
+        accelerator,
+        devices,
+        cfg_path=cfg_path,
+    )
+
+
+@pytest.mark.parametrize("accelerator, devices", accelerator_devices_pairs())
+def test_inference_dataframe_without_boxes(
+    accelerator: str, devices: int, features_from_df_command_and_paths: Tuple[str, Path, Path]
+) -> None:
+    command, features_path, cfg_path = features_from_df_command_and_paths
+    command = command.format(df_name=MOCK_DATASET_PATH / "df.csv")
+    run(command, accelerator, devices, cfg_path=cfg_path)
+
+
+@pytest.mark.parametrize("accelerator, devices", accelerator_devices_pairs())
+def test_inference_compare_features_from_df_and_path(
+    accelerator: str,
+    devices: int,
+    features_from_df_command_and_paths: Tuple[str, Path, Path],
+    features_from_path_command_and_paths: Tuple[str, Path, Path],
+) -> None:
+    command_df, features_df_path, cfg_df_path = features_from_df_command_and_paths
+    command_df = command_df.format(df_name=MOCK_DATASET_PATH / "df.csv")
+    command_w_path, features_w_path_path, cfg_w_path_path = features_from_path_command_and_paths
+    run(command_df, accelerator, devices, cfg_df_path)
+    run(command_w_path, accelerator, devices, cfg_w_path_path)
+
+    with features_df_path.open() as f, features_w_path_path.open() as f1:
```

Comment:
`f` and `f1` look meaningless if both are presented simultaneously, let's rename them at least to `f_df` and `f_folder`